### PR TITLE
Framework: Drop deprecations slated for 3.2 removal

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to remove deprecations slated for removal in the upcoming 3.2.0 release.

Specifically, this includes:

- `wp.data.withRehydratation` has been renamed to `wp.data.withRehydration`.
- The `wp.editor.ImagePlaceholder` component is removed. Please use `wp.editor.MediaPlaceholder` instead.
- `wp.utils.deprecated` function removed. Please use `wp.deprecated` instead.
- `getInserterItems`: the `allowedBlockTypes` argument was removed and the `parentUID` argument was added.
- `getFrecentInserterItems` selector removed. Please use `getInserterItems` instead.
- `getSupportedBlocks` selector removed. Please use `canInsertBlockType` instead.

See: https://wordpress.org/gutenberg/handbook/reference/deprecated/#3-2-0

__Testing instructions:__

Verify that there are no regressions in impacted behavior, and importantly that no references to deprecated behaviors exist in core code.

### CODE REVIEW

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks like `utils/deprecated.js` is slated to be removed but the file and function are still there: https://github.com/WordPress/gutenberg/blob/remove/3-2-deprecations/utils/deprecated.js

Otherwise looks good.

**Comment by jorgefilipecosta** [comment]
I did some grepping and testing and things seem fine. I think the only thing missing is the removal of the deprecated wrapper from /utils/deprecated.

**Comment by mcsf** [comment]
We should mention this step of checking for deprecations in the [release document](https://github.com/WordPress/gutenberg/blob/master/docs/reference/release.md).

**Review by tofumatt — APPROVED** [review]

**Comment by tofumatt** [comment]
Easily solved with a **Release Checklist** section, which totally belongs in the release docs.

**Comment by aduth** [comment]
That would work, but only if it's the _last_ step, not a pre-release step. Otherwise we risk reenforcing the idea that this be done at the time of a release, which I disagree with. 3.3 deprecations, for example,  should ideally be removed immediately after the upcoming 3.2 release.

**Comment by tofumatt** [comment]
Oh, wait, right. Okay! I will add...

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "gutenberg",
-	"version": "3.1.1",
+	"version": "3.2.0",
```
Comment (aduth) [inline package.json]: Sigh.. this wasn't meant to be included 😞 Only to help trigger the failing E2E locally.

Comment (tofumatt) [reply package.json]: Sorry I missed it too 😢

**Pull request merged**